apps/cowswap-frontend/src/legacy/state/orders/updaters/UnfillableOrdersUpdater.ts

```diff
@@ -144,7 +146,11 @@ export function UnfillableOrdersUpdater(): null {
         console.debug(`[UnfillableOrdersUpdater] Check order`, order)
 
         const currencyAmount = CurrencyAmount.fromRawAmount(order.inputToken, order.sellAmount)
-        const enoughBalance = hasEnoughBalanceAndAllowance({ account, amount: currencyAmount, balances })
+        const enoughBalance = hasEnoughBalanceAndAllowance({
+          account,
+          amount: currencyAmount,
+          balances: balancesRef.current,
```

Comment:
Now balances uses a ref to not re-render the function that often, but always use the latest version of the balances